exchange_calendars/exchange_calendar.py

```diff
@@ -532,71 +575,253 @@ def special_offsets_adhoc(self):
         """
         return []
 
-    # -----
+    # ------------------------------------------------------------------
+    # -- NO method below this line should be overriden on a subclass! --
```

Comment:
We should enforce this via an `__init_subclass__` checking this.

Reply:
I wouldn't want to impede users doing what they want with their own calendars. I've added a test `test_base_integrity` to ensure the integrity of the pre-defined ones.